src/pmc_simulation.jl

```diff
@@ -21,42 +21,57 @@ mutable struct PMCSimulation
 end
 
 function _set_up_starting_vectors(
-    start_at, n_replicas, n_spectral, style, initiator, threading, copy_vectors
+    ham, start_at, n_replicas, n_spectral, style, initiator, threading, copy_vectors
 )
     if start_at isa AbstractMatrix && size(start_at) == (n_replicas, n_spectral)
         if eltype(start_at) <: Union{AbstractDVec, RMPI.MPIData} # already dvecs
             if copy_vectors
-                return SMatrix{n_spectral, n_replicas}(deepcopy(v) for v in start_at)
+                return SMatrix{n_replicas, n_spectral}(deepcopy(v) for v in start_at)
             else
-                return SMatrix{n_spectral, n_replicas}(v for v in start_at)
+                return SMatrix{n_replicas, n_spectral}(v for v in start_at)
             end
         else
-            return SMatrix{n_spectral, n_replicas}(
+            return SMatrix{n_replicas, n_spectral}(
                 default_starting_vector(sa; style, initiator, threading) for sa in start_at
             )
         end
+    elseif n_spectral > 1
+        basis = build_basis(ham; minimum_size=n_spectral*5)
```

Comment:
Hmm, I'm not sure whether the factor 5 is reasonable here. It seems a large. The larger this matrix, the better the starting vectors will be, but as soon as these numbers become larger, then `eigvecs` can consume a lot of resources. Maybe 2 is more reasonable?

```suggestion
        basis = build_basis(ham; minimum_size=n_spectral*2)
```
We could also think about allowing the user to set the `minimum_size` by making it a keyword argument in `ProjectorMonteCarloProblem` that is passed through. What do you think?

Reply:
Another comment I have here is that the current code ignores the value of `start_at` in this branch. There might be value in allowing to pass an `allowed_address_type`, which can then be passed as a second argument into `build_basis`.

Reply:
Ok, I've set the default to a factor of 2, and made `minimum_size` a keyword argument in `ProjectorMonteCarloProblem`. I think that will be useful to allow the option of starting with better approximations to the eigenvectors.

At the moment `build_basis` just uses the `starting_address` of the Hamiltonian, is there a situation where something else should be used?

Reply:
```quote
At the moment build_basis just uses the starting_address of the Hamiltonian, is there a situation where something else should be used?
```
I'm thinking of a momentum-conserving momentum-space Hamiltonian (e.g. `HubbardMom1D`) where you could access blocks with different total momentum simply by starting from a different address (which would be chosen to have the total momentum that you want to consider). This could be done by creating a new Hamiltonian with a different starting address, of course, but there might be some value of being able to pass that in later.

The other use case is that you've created a Hamiltonian (possibly strongly correlated) but the `starting_address` is not a very good starting point. What is a good starting point may depend on the value of an interaction strength parameter, for example.